pkg/util/fuzzystrmatch/leven.go

```diff
@@ -10,13 +10,7 @@
 
 package fuzzystrmatch
 
-// LevenshteinDistanceWithCost calculates the Levenshtein distance between
-// source and target. The distance is calculated using the given cost metrics
-// for each of the three possible edit operations.
-// Adapted from the 'Iterative with two matrix rows' approach within
-// https://en.wikipedia.org/wiki/Levenshtein_distance. This approach provides us
-// with O(n^2) time complexity and O(n) space complexity.
-func LevenshteinDistanceWithCost(source, target string, insCost, delCost, subCost int) int {
+func levenshtein(source, target string, insCost, delCost, subCost, maxDistance int) int {
```

Comment:
i'm trying to decipher whether re-use the same function for all the levenshtein function variations?

to me there's a lot of extra logic here, i am wondering whether we should have the `LevenshteinDistanceWIthCost` be the same as before, and this `LessEqual` have the additional logic needed here. 

also, where was this algorithm based from? a hyperlink if you had inspiration from somewhere else would be great!